src/parameterizations/vertical/MOM_diabatic_driver.F90

```diff
@@ -1208,11 +1208,19 @@ subroutine diabatic_ALE(u, v, h, tv, Hml, fluxes, visc, ADp, CDp, dt, Time_end,
                                  CS%KPP_buoy_flux, CS%KPP_temp_flux, CS%KPP_salt_flux)
 
     ! The KPP scheme calculates boundary layer diffusivities and non-local transport.
-    call KPP_compute_BLD(CS%KPP_CSp, G, GV, US, h, tv%T, tv%S, u, v, tv, &
-                         fluxes%ustar, CS%KPP_buoy_flux, Waves=Waves)
+    if ( associated(fluxes%lamult) ) then
```

Comment:
I agree that these changes make sense, but would it also make sense to make similar changes to the other calls to `KPP_compute_BLD()` and `KPP_calculate()` inside of `diabatic_ALE_legacy()` at lines 631-635 of this file and inside of `layered_diabatic()` at lines 1805-1809 of this file?

Reply:
Agreed! I'll make the same changes for the remaining calls.

Reply:
Thanks, @Hallberg-NOAA. @alperaltuntas incorporated these changes in https://github.com/NOAA-GFDL/MOM6/pull/1491/commits/5f8c446d91d6942a78fc31322a1c690ec7e28c79